lib/sprockets/utils/gzip.rb

```diff
@@ -2,11 +2,50 @@
 module Sprockets
   module Utils
     class Gzip
+
+      # Private: Generates a gzipped file based off of reference asset.
+      #
+      #     ZlibArchiver.call(file, source, mtime)
+      #
+      # Compresses a given `source` using stdlib Zlib algorithm
+      # writes contents to the `file` passed in. Sets `mtime` of
+      # written file to passed in `mtime`
+      module ZlibArchiver
+        def self.call(file, source, mtime)
+          gz = Zlib::GzipWriter.new(file, Zlib::BEST_COMPRESSION)
+          gz.mtime = mtime
+          gz.write(source)
+          gz.close
+
+          File.utime(mtime, mtime, file.path)
+        end
+      end
+
+      # Private: Generates a gzipped file based off of reference asset.
+      #
+      #     ZopfliArchiver.call(file, source, mtime)
+      #
+      # Compresses a given `source` using the zopfli gem
+      # writes contents to the `file` passed in. Sets `mtime` of
+      # written file to passed in `mtime`
+      module ZopfliArchiver
+        def self.call(file, source, mtime)
+          compressed_source = Autoload::Zopfli.deflate(source, format: :gzip, mtime: mtime)
```

Comment:
Do we need to autoload this? We only register this exporter if the constant is defined so at this point it was already loaded.